pkg/auth/providers/common/usermanager.go

```diff
@@ -238,21 +240,38 @@ func (m *userManager) EnsureUser(principalName, displayName string) (*v3.User, e
 		return nil, err
 	}
 
-	logrus.Infof("Creating globalRoleBinding for %v", created.Name)
-	_, err = m.globalRoleBindings.Create(&v3.GlobalRoleBinding{
-		ObjectMeta: v1.ObjectMeta{
-			GenerateName: "globalrolebinding-",
-		},
-		UserName:       created.Name,
-		GlobalRoleName: "user",
-	})
+	logrus.Infof("Creating globalRoleBindings for %v", created.Name)
+	err = m.createNewUsersBindings(created.Name)
 	if err != nil {
 		return nil, err
 	}
 
 	return created, nil
 }
 
+func (m *userManager) createNewUsersBindings(name string) error {
```

Comment:
as discussed, we need to use the same cluster/proejct annotation based logic here for user